### ISSUE TITLE
Uncaught Error: The result of this StaticQuery could not be fetched.

### ISSUE THREAD

**Issue by jack0wsky** [gatsbyjs/gatsby#26563]
## Description

I have deployed site through Vercel and I'm facing following error. 

```
Uncaught Error: The result of this StaticQuery could not be fetched.

This is likely a bug in Gatsby and if refreshing the page does not fix it, please open an issue in https://github.com/gatsbyjs/gatsby/issues
    at h (gatsby-browser-entry.js:77)
    at e.default (products.js:15)
    at Ki (react-dom.production.min.js:153)
    at Fa (react-dom.production.min.js:175)
    at vo (react-dom.production.min.js:263)
    at cu (react-dom.production.min.js:246)
    at ou (react-dom.production.min.js:246)
    at Zo (react-dom.production.min.js:239)
    at react-dom.production.min.js:123
    at KqkS.t.unstable_runWithPriority (scheduler.production.min.js:19)
```
Error shows when I click on `Products` page or even `Contact` page where isn't static query declared. This error didn't exist while developing. Here's `products.js` file:

```
import { useStaticQuery, graphql } from 'gatsby'
import Product from '../components/product/product'

const Products = () => {
  const {
    allShopifyProduct: { edges },
  } = useStaticQuery(graphql`
    query Products {
      allShopifyProduct {
        edges {
          node {
            title
            productType
            shopifyId
            handle
            images {
              id
              localFile {
                childImageSharp {
                  fluid {
                    srcWebp
                    tracedSVG
                    base64
                    srcSetWebp
                  }
                  fixed {
                    src
                  }
                }
              }
            }
            variants {
              sku
              price
              title
              shopifyId
            }
          }
        }
      }
    }
  `)
  return (
    <Wrapper>
      <ProductsGrid>
        {edges.map(({ node }) => {
          return <Product key={node.shopifyId} product={node} />
        })}
      </ProductsGrid>
    </Wrapper>
  )
}

export default Products
```
I added `query Products` because I thought it'll help but it didn't.

### Expected result

website should work in production

### Actual result

some problems with static query I guess

 System:
    OS: macOS 10.15.6
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.14.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 84.0.4147.125
    Firefox: 79.0
    Safari: 13.1.2
  npmPackages:
    gatsby: ^2.17.4 => 2.24.10
    gatsby-cli: ^2.12.62 => 2.12.65
    gatsby-image: ^2.2.29 => 2.4.13
    gatsby-plugin-google-analytics: ^2.0.13 => 2.3.13
    gatsby-plugin-layout: ^1.0.11 => 1.3.10
    gatsby-plugin-manifest: ^2.2.23 => 2.4.20
    gatsby-plugin-offline: ^3.0.16 => 3.2.20
    gatsby-plugin-react-helmet: ^3.1.13 => 3.3.10
    gatsby-plugin-root-import: ^2.0.5 => 2.0.5
    gatsby-plugin-sharp: ^2.2.32 => 2.6.21
    gatsby-plugin-styled-components: ^3.3.10 => 3.3.10
    gatsby-plugin-transition-link: ^1.20.2 => 1.20.2
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4
    gatsby-source-filesystem: ^2.1.33 => 2.3.22
    gatsby-source-graphql: ^2.6.2 => 2.6.2
    gatsby-source-shopify: ^3.2.24 => 3.2.24
    gatsby-transformer-sharp: ^2.3.0 => 2.5.11
  npmGlobalPackages:
    gatsby-cli: 2.12.87

**Comment by pieh** [comment]
Could you try updating `gatsby` and `gatsby-plugin-offline` to latest versions? There were couple of fixes related to static queries in those packages. If that doesn't help - could you share a link to your project (preferably) or to site deployed to Vercel so we could try debugging the runtime and see what's going wrong there?

**Comment by barbalex** [comment]
I just had the exact same error - ONLY ON PRODUCTION!

The error did not appear in dev.

I then realized that the only place I used a static query (via the useStaticQuery hook) could just as well accept a hardcoded value. After changing that the next update to production did not error any more. Phew!

**Comment by Jaikant** [comment]
Facing the same issue.

gatsby version 2.24.53

and I do not have gatsby-plugin-offline installed.

**Comment by wardpeet** [comment]
@jack0wsky @barbalex @Jaikant could you please create a reproduction for us or share your repository. We haven't been able to reproduce this issue. You can always sent me an email at ward@gatsbyjs.com

**Comment by wardpeet** [comment]
I've had a call with a customer and they were using cloudfront CDN. Everything was cached, please make sure you do not cache HTML, page-data folder! See https://www.gatsbyjs.com/docs/caching/

**Comment by barbalex** [comment]
Being a complete noob in these matters: Does https://vercel.com/docs/edge-network/caching mean that that happened in my case too?

**Comment by wardpeet** [comment]
@barbalex I'm not sure, if I believe their source code it should have the correct cache-control:
https://github.com/vercel/vercel/blob/35fd7b5f9c628f9304cd9d2dcce9446ec4a2c1c7/packages/now-static-build/src/frameworks.ts#L72-L76

Perhaps it's something else, could you send me your url? Perhaps through ward@gatsbyjs.com if you don't want to share it publicly?

**Comment by barbalex** [comment]
@wardpeet 
It was this repository: https://github.com/FNSKtZH/ae2. If you check it out at commit 5673e1d282e67e0c52ffff116e27c41d089e912e you get the version that errored.

The repository is open source. Only the data in the postgresql database backing it up is not.

I just realized that this is the only one of my projects that I do _not_ deploy via vercel (sorry for the noise!) but rather via the docker-compose.yml in the backend directory that is run in a docker droplet of digitalocean.com.

That docker backend uses caddy to serve the html folder. As I never spent any thought on caching and do not understand how that is done by caddy it is possible that it's caching was too aggressive. This may not have been a good idea but at the time I had just tried out building the backend via docker (migrating from more traditional, self configured droplets) and I simply gave this way to do it a shot. And it did work a few weeks before the incident and since.

**Comment by wardpeet** [comment]
well what this issue is showing is that we should make people more aware of caching and/or update static-query to contain a timestamp

**Comment by einarpersson** [comment]
I just got the same error message running Gatsby in production on [Vercel](https://vercel.com): `StaticQuery could not be fetched...`

The latest commit was totally harmless, I just removed a file which was not references anywhere. And after doing another change to `README.md` it suddenly began to work in production.

**This is super confusing for someone who is new to Gatsby**

**Comment by kbrock84** [comment]
I am currently getting this error. Initially I only noticed it after our build server pushed it to our staging environment. It took me forever, but I finally reproduced it locally with the following:

1. go back to previous commit
2. delete `node_modules`
3. `npm install`

4. Then I ran into a separate [issue with babel](https://github.com/babel/babel/issues/11216) so I ran `npm upgrade gatsby` which also upgraded the `@babel` dependencies.
 
5. Finally `gatsby build && gatsby serve`

I was able to view the exact error locally, which is causing some of the data to not show on the page. It feels like a dependency issue.

**Comment by kbrock84** [comment]
Ok, so building using `gatsby build --no-uglify` gave me an idea of where the error was coming from. Changed that form `useStaticQuery` to a `StaticQuery` component and the error went away.

The site looks great now EXCEPT there is a random "Loading (StaticQuery)" inside a div on the page. This has to be related, but at least it's not breaking anything now.

Again, this only happens on the production build.

**Comment by claygiffin** [comment]
I am running into this same issue. It happens with nearly every production build that I push to Netlify. I have not been able to reproduce the issue locally, and there are no build errors. 

On my production site, the only solution I've found is to clear the cache and rebuild after each site update.

My current dependencies are:
```
"dependencies": {
    "babel-plugin-styled-components": "^1.11.1",
    "bowser": "^2.11.0",
    "gatsby": "^2.24.78",
    "gatsby-image": "^2.4.21",
    "gatsby-plugin-anchor-links": "^1.1.1",
    "gatsby-plugin-facebook-pixel": "^1.0.5",
    "gatsby-plugin-google-analytics": "^2.3.14",
    "gatsby-plugin-manifest": "^2.4.34",
    "gatsby-plugin-netlify": "^2.3.19",
    "gatsby-plugin-netlify-cache": "^1.2.0",
    "gatsby-plugin-react-helmet": "^3.3.14",
    "gatsby-plugin-react-svg": "^3.0.0",
    "gatsby-plugin-sharp": "^2.6.42",
    "gatsby-plugin-styled-components": "^3.3.14",
    "gatsby-source-filesystem": "^2.3.34",
    "gatsby-transformer-sharp": "^2.5.17",
    "google-map-react": "^2.1.8",
    "lodash": "^4.17.20",
    "modern-normalize": "^1.0.0",
    "polished": "^4.0.2",
    "prop-types": "^15.7.2",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "react-google-maps": "^9.4.5",
    "react-helmet": "^6.1.0",
    "react-icons": "^3.11.0",
    "react-intersection-observer": "^8.29.0",
    "react-merge-refs": "^1.1.0",
    "react-player": "^2.6.2",
    "react-portal": "^4.2.1",
    "react-spring": "^8.0.27",
    "styled-components": "^5.2.0"
  },
  "devDependencies": {
    "@types/styled-components": "^5.1.4",
    "prettier": "2.1.2"
  },
```

**Comment by wardpeet** [comment]
🤔 I assumed it was caching headers or the platform not purging but it seems like we should add a hash to the file

**Comment by andreydrozd-smfe** [comment]
I too am experiencing this issue and it's definitely not a cache issue for me. I too do not experience this issue locally. 

I'm setting up a staging environment with no cache, just hosting the files on S3, and I am getting an undefined property error. It's like the query never ran. I empty the S3 bucket manually and redeploy, and the issue is still there.

My current dependencies are: 

```
"dependencies": {
    "@fortawesome/fontawesome-svg-core": "^1.2.22",
    "@fortawesome/free-brands-svg-icons": "^5.11.2",
    "@fortawesome/free-regular-svg-icons": "^5.11.2",
    "@fortawesome/free-solid-svg-icons": "^5.10.2",
    "@fortawesome/react-fontawesome": "^0.1.4",
    "@rhysforyou/gatsby-plugin-safari-site-icon": "^1.0.2",
    "@statuspage/status-widget": "^1.0.4",
    "babel-plugin-styled-components": "^1.10.6",
    "bootstrap": "^4.3.1",
    "crypto-browserify": "^3.12.0",
    "env-cmd": "^10.0.1",
    "gatsby": "^2.13.73",
    "gatsby-image": "^2.2.10",
    "gatsby-plugin-canonical-urls": "^2.1.18",
    "gatsby-plugin-facebook-pixel": "^1.0.3",
    "gatsby-plugin-favicon": "^3.1.6",
    "gatsby-plugin-google-analytics": "^2.1.23",
    "gatsby-plugin-lodash": "^3.1.8",
    "gatsby-plugin-manifest": "^2.2.6",
    "gatsby-plugin-offline": "^2.2.7",
    "gatsby-plugin-react-helmet": "^3.1.4",
    "gatsby-plugin-resolve-src": "^2.0.0",
    "gatsby-plugin-s3": "^0.3.2",
    "gatsby-plugin-sharp": "^2.2.13",
    "gatsby-plugin-sitemap": "^2.2.21",
    "gatsby-plugin-styled-components": "^3.1.11",
    "gatsby-plugin-web-font-loader": "^1.0.4",
    "gatsby-source-filesystem": "^2.1.11",
    "gatsby-source-prismic-graphql": "~3.4.0-beta.2",
    "gatsby-transformer-sharp": "^2.2.7",
    "js-cookie": "^2.2.1",
    "lodash": "^4.17.15",
    "memoize-one": "^5.1.1",
    "moment": "^2.24.0",
    "moment-timezone": "^0.5.27",
    "polished": "^3.4.1",
    "prismic-reactjs": "^1.0.1",
    "prop-types": "^15.7.2",
    "querystring": "^0.2.0",
    "react": "^16.9.0",
    "react-big-calendar": "^0.22.1",
    "react-cookie-consent": "^2.5.0",
    "react-dom": "^16.9.0",
    "react-helmet": "^5.2.1",
    "react-hooks-sse": "^1.0.0",
    "react-lazyload": "^2.6.5",
    "react-select": "^3.1.0",
    "react-slick": "^0.25.2",
    "react-string-replace": "^0.4.4",
    "react-super-responsive-table": "^5.1.1",
    "react-tooltip": "^3.11.1",
    "react-vis": "^1.11.7",
    "reactstrap": "^8.0.1",
    "slick-carousel": "^1.8.1",
    "smooth-scroll": "^16.1.0",
    "socket.io-client": "^2.3.0",
    "stickybits": "^3.7.1",
    "styled-components": "^4.3.2",
    "styled-tools": "^1.7.1",
    "whatwg-fetch": "^3.0.0"
  },
```

**Comment by claygiffin** [comment]
I _may_ have figured out what the issue is. I haven't run into this error for my last few builds now after making this change.

When I set up my new Gatsby project, I changed the case of the default components (layout.js and seo.js) to be UpperCamelCase (Layout.js, Seo.js). I know that git has issues with changing filename case, so I had to update my git.config to register these changes by running `git config core.ignorecase false`.

_However_, it appears that the original lowercase file was never completely removed. It didn't appear in my repository or local branch, but I noticed that when I staged my changes in VSCODE, both the lowercase and uppercase versions of the files were listed!

To fully remove the old versions of the files, I had to run:
```
git mv -f src/components/layout.js src/components/Layout.js &&
git mv -f src/components/seo.js src/components/Seo.js
```

Pushing that change _seems_ to have fixed the Static Query error.

**Comment by wardpeet** [comment]
Oh I'll try to use this as a reproduction 👍 Thanks!

**Comment by persocon** [comment]
any news? I tried all the solutions, none works :/ and I can't update any staticquery on the company website because it breaks EVERYTHING

**Comment by gerisztein** [comment]
Same issue as @persocon ☝️ 

> any news? I tried all the solutions, none works :/ and I can't update any staticquery on the company website because it breaks EVERYTHING
> 
> edit:
> `<StaticQuery />` works but `useStaticQuery` doesn't and I have a bunch of places that I use the `useStaticQuery` :/

**Comment by vincentleeuwen** [comment]
Same issue as @persocon and @gerisztein ☝️

> 
> > any news? I tried all the solutions, none works :/ and I can't update any staticquery on the company website because it breaks EVERYTHING
> > edit:
> > `<StaticQuery />` works but `useStaticQuery` doesn't and I have a bunch of places that I use the `useStaticQuery` :/

**Comment by Mehuge** [comment]
I have a reproducible case, out of the box.

```
npm i -g gatsby-cli
gatsby new mywebsite
cd mywebsite
gatsby develop
```

Then browse to http://localhost:8000/

gives the error.  See #27577 for details.

**Comment by claygiffin** [comment]
@Mehuge I was not able to reproduce the error locally following those steps.

**Comment by kbrock84** [comment]
I resolved this error, though not in a desirable way.

I removed this from my `gatsby-config.js`
```javascript
//gatsby-config.js
	siteMetadata: {
		title: `Site Title`,
		description: `Site Description`,
		author: `@siteauthor`,
	},
```

and just hard coded the values in the `seo.js` component and removed the graphql query.
```javascript
//seo.js
const site = {
	siteMetadata: {
		title: `Site Title`,
		description: `Site Description`,
		author: `@siteauthor`,
	},
};
```

And now I no longer have the error.

**Comment by Mehuge** [comment]
> @Mehuge I was not able to reproduce the error locally following those steps.

@claygiffin I have more info about my particular problem, and it is related to case. Having read you earlier comment about case of components, and having run gatsby develop from CMD prompt and it working, it turns out to be a cygwin case issue.

See, my projects are in `D:\DEV` but in cygwin I lazily `cd /d/dev` so there is a discrepancy in case in the root part of paths. I noticed these warnings 

```
warn The plugin "dev-404-page" created a page with a component path that doesn't match the casing of the actual file. This may work locally, but will break on systems which are case-sensitive, e.g. most CI/CD
pipelines.

page.component:     "D:/dev/website/mywebsite/.cache/dev-404-page.js"
path in filesystem: "D:/DEV/website/mywebsite/.cache/dev-404-page.js"
```

and

```
warn There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* D:\DEV\website\mywebsite\node_modules\prop-types\checkPropTypes.js
    Used by 2 module(s), i. e.
    D:\DEV\website\mywebsite\node_modules\react\cjs\react.development.js
* D:\dev\website\mywebsite\node_modules\prop-types\checkPropTypes.js
    Used by 3 module(s), i. e.
    D:\dev\website\mywebsite\node_modules\react\cjs\react.development.js
```

and 

```
warn The GraphQL query in the non-page component "D:/dev/website/mywebsite/src/pages/using-typescript.tsx" will not be run.
Exported queries are only executed for Page components. It's possible you're
trying to create pages in your gatsby-node.js and that's failing for some
reason.

If the failing component(s) is a regular component and not intended to be a page
component, you generally want to use a <StaticQuery> (https://gatsbyjs.org/docs/static-query)
instead of exporting a page query.

If you're more experienced with GraphQL, you can also export GraphQL
fragments from components and compose the fragments in the Page component
query and pass data down into the child component — https://graphql.org/learn/queries/#fragments
```

I only took notice of the warnings after realising that when run from CMD prompt it doesn't generate any warnings.

**Comment by bpod** [comment]
Figured it out. For my issue anyway, but it seems as if other peoples issue may be stemming from the same kind of symptoms.

For me it was `emotion-cache`. 

In a Netlify production build, it would fail the first time a page would load while using the `useStaticQuery` hook. If you would refresh, it would not reproduce the error.. so only on first load. This got me thinking it might have something to do with caching strategy. I removed my custom implementation of emotion-cache and now everything appears to be just fine.

**Comment by bpod** [comment]
> Figured it out. For my issue anyway, but it seems as if other peoples issue may be stemming from the same kind of symptoms.
> 
> For me it was `emotion-cache`.
> 
> In a Netlify production build, it would fail the first time a page would load while using the `useStaticQuery` hook. If you would refresh, it would not reproduce the error.. so only on first load. This got me thinking it might have something to do with caching strategy. I removed my custom implementation of emotion-cache and now everything appears to be just fine.
> 
> This is a rather unique case, but hopefully it helps others narrow this down.

I Lied... Still an issue. :(

**Comment by claygiffin** [comment]
@bpod At any point during your project did you change the case of any filenames?

**Comment by bpod** [comment]
> @bpod At any point during your project did you change the case of any filenames?

Nope.. I thought of that too... but no, been consistent with that.

**Comment by persocon** [comment]
any news on that? 😭

**Comment by honmanyau** [comment]
I ran into the same issue earlier and it was a case error in the **Importing** component, but not the **Imported** component where the static query is made:

```jsx
// index.js
import React from 'react';

import Header from '../components/Header'

// ...
```

```jsx
// header.js
import React from 'react';
import { Link, graphql, useStaticQuery } from 'gatsby';

const Header = () => {
  const data = useStaticQuery(graphql`query {
    site {
      siteMetadata {
        title
      }
    }
  }`);

  // ...
};

export default Header;
```

Using the example above, the solution for me was to simply change the filename of the relevant import in `index.js` to match the filename of the imported component:

```jsx
// index.js
import React from 'react';

import Header from '../components/header'

// ...
```

I initially attempted `gatsby clean`, removed `node_modules` and reinstalled them but that didn't work for me. Although I doubt that it was the problem to begin with, because after playing around for a while the only way I could cause the error to be thrown was unmatching cases between the `import` statement and the filename of the imported component.

Also not sure if it's worth mentioning, but might as well throw it out there: in `.cache/gatsby-browser-entry.js`, it appears that `React.useContext(StaticQueryContext)` returns an empty object when the aforementioned case mismatch exists, which is what lead to the error being thrown in the first place.

Note that this is just what happened in my particularly case, but I hope it helps!

**Comment by observerzy** [comment]
same problem
any new to solve it?

**Comment by caesarsol** [comment]
For anyone hitting the `Loading (StaticQuery)` problem as described by @kbrock84, I found the solution in https://github.com/gatsbyjs/gatsby/issues/26038#issuecomment-664987789:

Remove `gatsby-plugin-offline` and add `gatsby-plugin-remove-serviceworker` from the Gatsby plugins (you'll need to `npm install` that).

**Comment by caesarsol** [comment]
More information on my findings: the cause seems to be the `StaleWhileRevalidate` strategy in fetching `page-data.json` files:
https://github.com/gatsbyjs/gatsby/blob/e4fd0ad8cda56996f42e0a2b9fd6b580823289c9/packages/gatsby-plugin-offline/src/gatsby-node.js#L144-L145

This makes every request to `page-data.json` return with an old version of the file, which contains the incorrect `staticQueryHashes`.
If you compare the requests in the errored page and after a refresh, you'll see the content of the relative `page-data.json` will have different `staticQueryHashes`, being the latest correct.

This also explains the apparent random nature of the error: it happens only after a query changes, and the ServiceWorker in the browser still has the previous version cached.

I am unsure on the best solution when one wants to keep the offline cache functionality:
 - Maybe changing the offline strategy from [StaleWhileRevalidate](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#stale-while-revalidate) to [NetworkFirst](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#network_first_network_falling_back_to_cache)? 
 - Or maybe the [Gatsby cache loader logic](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/loader.js#L272) fetching the chunks is wrong somewhere?
 - Another possibility is that I'm completely off-road and the right thing to do is check the cache-related headers of the JSON files, as [suggested here](https://github.com/gatsbyjs/gatsby/issues/26038#issuecomment-734712785), but I tried it with `npx check-gatsby-caching` and it's all green.

If someone has some ideas please get in touch!

**Comment by timurc** [comment]
@LekoArts asked my to add my reproduction. Here it is: 

### Steps to reproduce

I made a branch that at least _at my machine_ reproduces the error: https://github.com/voellig-ohne/timobecker/tree/reproduce-issue

Clone into repo, checkout out `reproduce-issue`, `npm i`,  `npm start`, open in browser

Things I tried:

* update all dependencies to the latest version
* delete `node_modules`, `.cache`
* check for casing errors
* fresh checkout in a different folder, just to be sure..

My original issue:  https://github.com/gatsbyjs/gatsby/issues/28537

**Comment by persocon** [comment]
im still having this issue and it only happens in prod 🤷 couldn't figure out yet how to solve, im just removing all the bullshit staticquery from it and it works 🤷‍♀️

**Comment by caesarsol** [comment]
@persocon I had the same exactly. Do you have gatsby-plugin-offline active?

**Comment by persocon** [comment]
@caesarsol yes I do have, did removing it solved for you?

**Comment by adamfaryna** [comment]
I don't use gatsby-plugin-offline, but I have the same problem with `useStaticQuery` and `StaticQuery`.

**Comment by caesarsol** [comment]
@persocon yes it did, please read my messages above.

**Comment by lattespirit** [comment]
> I _may_ have figured out what the issue is. I haven't run into this error for my last few builds now after making this change.
> 
> When I set up my new Gatsby project, I changed the case of the default components (layout.js and seo.js) to be PascalCase (Layout.js, SEO.js). I know that git by default ignores changes in filename case, so I had to update my git.config to register these changes by running `git config core.ignorecase false`.
> 
> _However_, it appears that the original lowercase file was never completely removed. It didn't appear in my repository or local branch, but I noticed that when I staged my changes in VSCODE, both the lowercase and uppercase versions of the files were listed!
> 
> To fully remove the old versions of the files, I had to run:
> 
> ```
> git mv -f src/components/layout.js src/components/Layout.js &&
> git mv -f src/components/seo.js src/components/SEO.js
> ```
> 
> Pushing that change seems to have fixed the Static Query error.

Thanks, you saved my day.

**Comment by MarekMaras** [comment]
I too am experiencing this issue it's not a cache issue for me.
Locally it works.
As soon as I deploy it to Netlify and open up my main domain, it shows this errors:

Error: The result of this StaticQuery could not be fetched.

```
This is likely a bug in Gatsby and if refreshing the page does not fix it, please open an issue in https://github.com/gatsbyjs/gatsby/issues
    at m (app-932771afe1ec633c2149.js:2)
    at O (component---src-pages-index-js-7077fee428ff1b64d194.js:1)
    at Ki (framework-305b3707783ccc9d7ca6.js:2)
    at vo (framework-305b3707783ccc9d7ca6.js:2)
    at cu (framework-305b3707783ccc9d7ca6.js:2)
    at ou (framework-305b3707783ccc9d7ca6.js:2)
    at Zo (framework-305b3707783ccc9d7ca6.js:2)
    at qo (framework-305b3707783ccc9d7ca6.js:2)
    at Du (framework-305b3707783ccc9d7ca6.js:2)
    at framework-305b3707783ccc9d7ca6.js:2
Za @ framework-305b3707783ccc9d7ca6.js:2

vTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/webpack-runtime-86bcfb54a9e4c0e2c403.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/app-932771afe1ec633c2149.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/framework-305b3707783ccc9d7ca6.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/4dbd8774c93329dba068f4828d4180636febe693-4865a04b1f4372fe8150.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/component---src-pages-index-js-7077fee428ff1b64d194.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/component---cache-caches-gatsby-plugin-offline-app-shell-js-ad6431e4664bcf916d19.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://www.marek-maras.de/cb1608f2-8597d708199ed60075bc.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

The Netlify Preview Deploy URL works like a charm so I guess, it's as soon as it switches to Production, those errors come up.

Running on **NODE_VERSION = v12** on Netlify because of an bug/error in postcss for tailwind v2.
Same occured on v15.4 and v15.5

`  "dependencies": {
    "@emotion/react": "^11.1.4",
    "@emotion/styled": "^11.0.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.32",
    "@fortawesome/free-solid-svg-icons": "^5.15.1",
    "@fortawesome/react-fontawesome": "^0.1.14",
    "@sanity/block-content-to-react": "^2.0.7",
    "@tailwindcss/forms": "^0.2.1",
    "autoprefixer": "^10.1.0",
    "cross-env": "^7.0.3",
    "dotenv": "^8.2.0",
    "esm": "^3.2.25",
    "fontsource-muli": "^4.0.0",
    "gatsby": "^2.29.2",
    "gatsby-cli": "^2.16.2",
    "gatsby-image": "^2.8.0",
    "gatsby-link": "^2.8.0",
    "gatsby-plugin-cookiehub-banner": "^0.1.0",
    "gatsby-plugin-manifest": "^2.9.1",
    "gatsby-plugin-netlify": "^2.8.0",
    "gatsby-plugin-nprogress": "^2.7.0",
    "gatsby-plugin-offline": "^3.7.1",
    "gatsby-plugin-postcss": "^3.4.0",
    "gatsby-plugin-react-helmet": "^3.7.0",
    "gatsby-plugin-robots-txt": "^1.5.5",
    "gatsby-plugin-sharp": "^2.11.2",
    "gatsby-plugin-sitemap": "^2.9.0",
    "gatsby-plugin-styled-components": "^3.7.0",
    "gatsby-plugin-typography": "^2.9.0",
    "gatsby-source-filesystem": "^2.8.1",
    "gatsby-source-sanity": "^6.0.4",
    "gatsby-transformer-sharp": "^2.9.0",
    "install": "^0.13.0",
    "netlify-lambda": "^2.0.2",
    "npm": "^6.14.10",
    "polished": "^4.0.5",
    "postcss": "^8.2.1",
    "postcss-cli": "^8.3.1",
    "precss": "^4.0.0",
    "prop-types": "^15.7.2",
    "react": "^17.0.1",
    "react-cookie-consent": "^6.2.1",
    "react-dom": "^17.0.1",
    "react-helmet": "^6.1.0",
    "react-router-dom": "^5.2.0",
    "react-router-hash-link": "^2.3.1",
    "react-scroll": "^1.8.1",
    "react-typography": "^0.16.19",
    "styled-components": "^5.2.1",
    "tailwindcss": "^2.0.2",
    "twin.macro": "^2.0.7",
    "typography": "^0.16.19"
  },
  "devDependencies": {
    "@babel/plugin-proposal-decorators": "^7.12.12",
    "@emotion/babel-plugin-jsx-pragmatic": "^0.1.5",
    "@hot-loader/react-dom": "^17.0.1",
    "@tailwindcss/custom-forms": "0.2.1",
    "babel-eslint": "10.1.0",
    "cssnano": "4.1.10",
    "eslint": "7.16.0",
    "eslint-loader": "3.0.4",
    "eslint-plugin-import": "2.22.1",
    "eslint-plugin-react": "7.21.5",
    "gatsby-plugin-eslint": "2.0.8",
    "prettier": "2.2.1"
  },`

**Comment by jm27** [comment]
Hi everyone, as suggested in another thread deleting the node_modules folder and reinstalling using npm install, fix this issue for me. I would like to know more about why it fix it? and what is causing the issue? Great product btw thanks!

**Comment by esotericjules** [comment]
> I ran into the same issue earlier and it was a case error in the **Importing** component, but not the **Imported** component where the static query is made:
> 
> ```js-jsx
> // index.js
> import React from 'react';
> 
> import Header from '../components/Header'
> 
> // ...
> ```
> 
> ```js-jsx
> // header.js
> import React from 'react';
> import { Link, graphql, useStaticQuery } from 'gatsby';
> 
> const Header = () => {
>   const data = useStaticQuery(graphql`query {
>     site {
>       siteMetadata {
>         title
>       }
>     }
>   }`);
> 
>   // ...
> };
> 
> export default Header;
> ```
> 
> Using the example above, the solution for me was to simply change the filename of the relevant import in `index.js` to match the filename of the imported component:
> 
> ```js-jsx
> // index.js
> import React from 'react';
> 
> import Header from '../components/header'
> 
> // ...
> ```
> 
> I initially attempted `gatsby clean`, removed `node_modules` and reinstalled them but that didn't work for me. Although I doubt that it was the problem to begin with, because after playing around for a while the only way I could cause the error to be thrown was unmatching cases between the `import` statement and the filename of the imported component.
> 
> Also not sure if it's worth mentioning, but might as well throw it out there: in `.cache/gatsby-browser-entry.js`, it appears that `React.useContext(StaticQueryContext)` returns an empty object when the aforementioned case mismatch exists, which is what lead to the error being thrown in the first place.
> 
> Note that this is just what happened in my particularly case, but I hope it helps!
> 
> **Edit:** I forgot to mention that I'm using macOS (10.15.7, case-insensitive).

This worked for me.

**Comment by momosetti** [comment]
I faced the same issue on Gatsby `2.31.1` and I downgrade it to `2.24.79` without using `gatsby-plugin-offline`, it works well.

**Comment by lanzoninicola** [comment]
The solution posted by @esotericjules works also for me.

**Comment by barbalex** [comment]
I am now encountering this issue in different ways:

- When using `StaticQuery` I geht this issue: https://github.com/gatsbyjs/gatsby/issues/26038, meaning: Instead of the component, `Loading (StaticQuery)` is shown.
- When using `useStaticQuery` the entire app blows up, returning `The result of this StaticQuery could not be fetched. This is likely a bug in Gatsby and if refreshing the page does not fix it, please open an issue in https://github.com/gatsbyjs/gatsby/issues`

This is only in production, so BAD 👿 

PLEASE SOLVE THIS ISSUE

It has been around for ages and many have endured it. 

Sorry for being so explicit but Gatsby being so big, this should have been solved a long time ago.

**Comment by hsaenz** [comment]
So, where we are with this? I made sure to:

- Not have invalid import names
- Update all my dependencies to latest
- Make sure git was not "hiding" renamed files from me

Nothing solved the issue and it would be real sad just to move away all my code from the hooks version.

This happens to me specifically by fetching images, it seems that the traced svg version from "GatsbyImageSharpFluid_tracedSVG" gets stuck at some point.

Also wondering if it has something to do with cache handling, as this will never happen to me when doing a hard reaload. 

Site example: https://gmdlatinoamerica.netlify.app/en/ (be sure to navigate between brand pages and the HP, you'll get the console error eventually and some images will just not complete loading).

**Comment by llamerr** [comment]
Same problem. After replacing `useStaticQuery` with `<StaticQuery />` it started working for me. Removing `node_modules` and `package-lock.json` did not help.

**Comment by endymion1818** [comment]
Could be related to https://github.com/gatsbyjs/gatsby/issues/29416

**Comment by endymion1818** [comment]
I've seen this bug too, not running locally but in production. In the pipeline it was built with Gatsby v2.30
```
  System:
    OS: macOS 10.15.7
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.19.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.0.3 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.150
    Edge: 88.0.705.63
    Firefox: 85.0.1
    Safari: 14.0.2
  npmPackages:
    gatsby: ^2.1.20 => 2.25.0
    gatsby-image: ^2.5.0 => 2.9.0
    gatsby-plugin-google-analytics: ^2.1.29 => 2.9.0
    gatsby-plugin-google-tagmanager: ^2.1.18 => 2.9.0
    gatsby-plugin-mdx: ^1.2.15 => 1.8.0
    gatsby-plugin-polyfill-io: ^1.1.0 => 1.1.0
    gatsby-plugin-react-helmet: ^3.1.10 => 3.8.0
    gatsby-plugin-react-helmet-canonical-urls: ^1.2.0 => 1.4.0
    gatsby-plugin-remove-serviceworker: ^1.0.0 => 1.0.0
    gatsby-plugin-remove-trailing-slashes: ^2.2.1 => 2.8.0
    gatsby-plugin-robots-txt: ^1.3.0 => 1.5.5
    gatsby-plugin-sharp: ^2.3.4 => 2.12.0
    gatsby-plugin-sitemap: ^2.0.1 => 2.10.0
    gatsby-plugin-styled-components: ^3.0.0 => 3.8.0
    gatsby-plugin-typescript: ^2.0.3 => 2.10.0
    gatsby-source-apiserver: 2.1.5 => 2.1.5
    gatsby-source-datocms: ^2.4.0 => 2.5.7
    gatsby-source-filesystem: ^2.0.1 => 2.9.0
    gatsby-transformer-javascript-frontmatter: ^2.0.0 => 2.8.0
    gatsby-transformer-sharp: ^2.3.6 => 2.10.0
  npmGlobalPackages:
    gatsby-cli: 2.19.1
```

**Comment by llamerr** [comment]
> Same problem. After replacing `useStaticQuery` with `<StaticQuery />` it started working for me. Removing `node_modules` and `package-lock.json` did not help.

In my case problem did not occur during develop, and only happened upon build. In all other cases of `useStaticQuery` everything worked fine. Difference from other cases was that `query` itself was imported from another file.

**Comment by Qurus** [comment]
I just got the same issue, when I add to pngOptions: {quality: 70} to static query.

**Comment by GbalsaC** [comment]
Having the same issue, when running the app in "gatsby develop" it works find, also after build if I use "gatsby serve" it works fine as well
![image](https://user-images.githubusercontent.com/5886517/115073673-498ec400-9ebe-11eb-972f-9558a868a1b1.png)


But when I host it through Nginx, it breaks with static query issue:
![image](https://user-images.githubusercontent.com/5886517/115073737-5d3a2a80-9ebe-11eb-9417-b937d6344ee8.png)

**Comment by GbalsaC** [comment]
I've found this issue only happens when production is hosted on HTTPS.

Here's what I've tried:

- Local develop: works
- Local Gatsby Serve: works
- Nginx Static build on http: Works
- Nginx Gatsby Serve on http: Works
- Nginx Gatsby Serve on HTTPS: Fails (StaticQuery error)
- Nginx Static build on Https: Fails (StaticQuery error)
**- NGINX develop proxy on HTTPS: works** 

```json
"dependencies": {
    "gatsby": "^2.24.47",
    "gatsby-image": "^2.4.16",
    "gatsby-plugin-manifest": "^2.4.23",
    "gatsby-plugin-offline": "^3.2.23",
    "gatsby-plugin-react-helmet": "^3.3.10",
    "gatsby-plugin-sass": "^2.3.12",
    "gatsby-plugin-sharp": "^2.6.27",
    "gatsby-source-filesystem": "^2.3.24",
    "gatsby-transformer-sharp": "^2.5.13",
    }
```

Gatsby CLI version: 2.19.2

Here's the Image component triggering the StaticQuery Error

```typescript
import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import Img from "gatsby-image"

/*
 * This component is built using `gatsby-image` to automatically serve optimized
 * images with lazy loading and reduced file sizes. The image is loaded using a
 * `useStaticQuery`, which allows us to load the image from directly within this
 * component, rather than having to pass the image data down from pages.
 *
 * For more information, see the docs:
 * - `gatsby-image`: https://gatsby.dev/gatsby-image
 * - `useStaticQuery`: https://www.gatsbyjs.com/docs/use-static-query/
 */

interface imageProps {
  file: 'landingApi' | 'landingAmazon' | 'landingShopify' | 'landingIzettle' | 'landingML' | 'landingLaptopEcom' | 'landingLaptop' | 'amazon_logo' | 'majo' | 'fsb' | 'lite_banner' | 'apiBanner' | 'apiOne' | 'apiTwo' | 'apiThree' | 'apiTutorial' | 'shopifyIntro' | 'shopifyConfig' | 'shopifyStampOnline' | 'shopifyStampPage' | 'mlLogo' | 'mlIntro' | 'mlConfig' | 'mlMobileStamp' | 'mlTestimony';
}

const Image = ({ file }: imageProps) => {
  const data = useStaticQuery(graphql`
    query {
      landingApi: file(relativePath: { eq: "Ilustracion_API_2021.png" }) {
        childImageSharp {
          fluid(maxWidth: 750) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      landingAmazon: file(relativePath: { eq: "Amazon_logo_gray.png" }) {
        childImageSharp {
          fluid(maxWidth: 750) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      landingShopify: file(relativePath: { eq: "shopify.png" }) {
        childImageSharp {
          fluid(maxWidth: 750) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      landingIzettle: file(relativePath: { eq: "iZettle.png" }) {
        childImageSharp {
          fluid(maxWidth: 750) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      landingML: file(relativePath: { eq: "MercadoLibre.png" }) {
        childImageSharp {
          fluid(maxWidth: 750) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      landingLaptop: file(relativePath: { eq: "Mockup_1.png" }) {
        childImageSharp {
          fluid(maxWidth: 750) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      landingLaptopEcom: file(relativePath: { eq: "Mockup_2.png" }) {
        childImageSharp {
          fluid(maxWidth: 650) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      majo: file(relativePath: { eq: "ItoahMajo.jpg" }) {
        childImageSharp {
          fluid(maxWidth: 300) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      fsb: file(relativePath: { eq: "FSB_LOGO_1200x1200.png" }) {
        childImageSharp {
          fluid(maxWidth: 300) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      liteBanner: file(relativePath: { eq: "homeBanner.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      apiBanner: file(relativePath: { eq: "felipecirulo@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      apiFeatureOne: file(relativePath: { eq: "01-1@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      apiFeatureSecond: file(relativePath: { eq: "FPicon_dependencias-1@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      apiFeatureThree: file(relativePath: { eq: "03@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      apiTutorial: file(relativePath: { eq: "01@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:244){
            ...GatsbyImageSharpFluid
          }
        }
      }
      shopifyIntro: file(relativePath: { eq: "shopifyIntro@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:644){
            ...GatsbyImageSharpFluid
          }
        }
      }
      shopifyConfig: file(relativePath: { eq: "shopifyConfig@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:644){
            ...GatsbyImageSharpFluid
          }
        }
      }
      shopifyStampOnline: file(relativePath: { eq: "StampOnline.png" }) {
        childImageSharp {
          fluid (maxWidth:644){
            ...GatsbyImageSharpFluid
          }
        }
      }
      shopifyStampPage: file(relativePath: { eq: "StampPage.png" }) {
        childImageSharp {
          fluid (maxWidth:644){
            ...GatsbyImageSharpFluid
          }
        }
      }
      mlLogo: file(relativePath: { eq: "ML_logo2.png" }) {
        childImageSharp {
          fluid (maxWidth:244){
            ...GatsbyImageSharpFluid
          }
        }
      }
      mlIntro: file(relativePath: { eq: "IntroImage@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:644){
            ...GatsbyImageSharpFluid
          }
        }
      }
      mlConfig: file(relativePath: { eq: "config@2x.png" }) {
        childImageSharp {
          fluid (maxWidth:644){
            ...GatsbyImageSharpFluid
          }
        }
      }
      mlMobileStamp: file(relativePath: { eq: "mobileStamp2.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      mlTestimony: file(relativePath: { eq: "Testimony2.jpg" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
      amazonLogo: file(relativePath: { eq: "Amazon_logo.png" }) {
        childImageSharp {
          fluid (maxWidth:344){
            ...GatsbyImageSharpFluid
          }
        }
      }
  }
`)
  if (file === 'majo') {
    return (
      <Img
        fluid={data.majo.childImageSharp.fluid}
        alt="Testimonio Majo"
      />
    )
  }
  else if (file === 'landingApi') {
    return (
      <Img
        fluid={data.landingApi.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'landingAmazon') {
    return (
      <Img
        fluid={data.landingAmazon.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'landingShopify') {
    return (
      <Img
        fluid={data.landingShopify.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'landingML') {
    return (
      <Img
        fluid={data.landingML.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'landingIzettle') {
    return (
      <Img
        fluid={data.landingIzettle.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'landingLaptop') {
    return (
      <Img
        fluid={data.landingLaptop.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'landingLaptopEcom') {
    return (
      <Img
        fluid={data.landingLaptopEcom.childImageSharp.fluid}
        alt="Landing Laptop"
      />
    )
  }
  else if (file === 'amazon_logo') {
    return (
      <Img
        fluid={data.amazonLogo.childImageSharp.fluid}
        alt="Amazon Logo"
      />
    )
  }
  else if (file === 'fsb') {
    return (
      <Img
        fluid={data.fsb.childImageSharp.fluid}
        alt="Testimonio FSB"
      />
    )
  }
  else if (file === 'lite_banner') {
    return (
      <Img
        fluid={data.liteBanner.childImageSharp.fluid}
        alt="Home Banner"
      />
    )
  }
  else if (file === 'apiBanner') {
    return (
      <Img
        fluid={data.apiBanner.childImageSharp.fluid}
        alt="API Banner"
      />
    )
  }
  else if (file === 'apiOne') {
    return (
      <Img
        fluid={data.apiFeatureOne.childImageSharp.fluid}
        alt="API Banner"
      />
    )
  }
  else if (file === 'apiTwo') {
    return (
      <Img
        fluid={data.apiFeatureSecond.childImageSharp.fluid}
        alt="API Banner"
      />
    )
  }
  else if (file === 'apiThree') {
    return (
      <Img
        fluid={data.apiFeatureThree.childImageSharp.fluid}
        alt="API Banner"
      />
    )
  }
  else if (file === 'apiTutorial') {
    return (
      <Img
        fluid={data.apiTutorial.childImageSharp.fluid}
        alt="API Tutorial"
      />
    )
  }
  else if (file === 'shopifyIntro') {
    return (
      <Img
        fluid={data.shopifyIntro.childImageSharp.fluid}
        alt="Shopify Intro"
      />
    )
  }
  else if (file === 'shopifyConfig') {
    return (
      <Img
        fluid={data.shopifyConfig.childImageSharp.fluid}
        alt="Shopify Config"
      />
    )
  }
  else if (file === 'shopifyStampOnline') {
    return (
      <Img
        fluid={data.shopifyStampOnline.childImageSharp.fluid}
        alt="Shopify Stamp Online"
      />
    )
  }
  else if (file === 'shopifyStampPage') {
    return (
      <Img
        fluid={data.shopifyStampPage.childImageSharp.fluid}
        alt="Shopify Stamp Online"
      />
    )
  }
  else if (file === 'mlLogo') {
    return (
      <Img
        fluid={data.mlLogo.childImageSharp.fluid}
        alt="MercadoLibre Logo"
      />
    )
  }
  else if (file === 'mlIntro') {
    return (
      <Img
        fluid={data.mlIntro.childImageSharp.fluid}
        alt="MercadoLibre Logo"
      />
    )
  }
  else if (file === 'mlConfig') {
    return (
      <Img
        fluid={data.mlConfig.childImageSharp.fluid}
        alt="MercadoLibre Config"
      />
    )
  }
  else if (file === 'mlMobileStamp') {
    return (
      <Img
        fluid={data.mlMobileStamp.childImageSharp.fluid}
        alt="MercadoLibre Config"
      />
    )
  }
  else if (file === 'mlTestimony') {
    return (
      <Img
        fluid={data.mlTestimony.childImageSharp.fluid}
        alt="MercadoLibre Testimony"
      />
    )
  }

  /*

  if (!data?.placeholderImage?.childImageSharp?.fluid) {
    return <div>Picture not found</div>
  }
  */

  return <Img fluid={data.majo.childImageSharp.fluid} />
}

export default Image
```

And Gatsby Config

```javascript
module.exports = {
  siteMetadata: {
    title: `FiscalPOP - Facturación`,
    description: `FiscalPOP es facturación personal para el mercado moderno, integraciones con eCommerce y facturación personal moderna.`,
    author: `@fiscalpop`,
  },
  plugins: [
    `gatsby-plugin-react-helmet`,
    `gatsby-plugin-sass`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `gatsby-starter-default`,
        short_name: `starter`,
        start_url: `/`,
        background_color: `#663399`,
        theme_color: `#663399`,
        display: `minimal-ui`,
        icon: `src/images/icon-96x96.png`, // This path is relative to the root of the site.
      },
    },
    // this (optional) plugin enables Progressive Web App + Offline functionality
    // To learn more, visit: https://gatsby.dev/offline
    // `gatsby-plugin-offline`,
  ],
}
```

**Comment by ramki2486** [comment]
gatsby-config.js
```
{
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/assets`,
      },
    }
```
gatsby-version: V3
If your gatsby-source-filesystem path is ..../assets or ..../images, you should use a direct file name in that folder like this
```
const data = useStaticQuery(graphql`
    {
      file(relativePath: { eq: "profile.jpeg" }) {
        childImageSharp {
          gatsbyImageData(layout: FIXED)
        }
      }
    }
  `)
```

**Comment by RomanHotsiy** [comment]
In my case, I used `useStaticQuery` in html.js file and it crashed in production build only. Worked just fine in dev mode.

Changing `useStaticQuery` to `StaticQuery` fixes the issue!

**Comment by dilatedigital** [comment]
I'm having this error in my current Gatsby starter I'm using. I just updated the dependencies and now I'm getting this error.
```
Error in function useStaticQuery in ./.cache/gatsby-browser-entry.js:77
```

My repo is [here](https://github.com/dilatedigital/gatsby-wp-starter/tree/wp-main). To reproduce, just clone the repo and run gatsby develop.

**Comment by KyleAMathews** [comment]
As many people have noticed, getting the case wrong on import names can cause this problem if you're on OSX as Gatsby is case insensitive when setting up static queries but OSX isn't so if you have `import foo from "./foo` but the file is actually `Foo.js` — it'll seem to work except for the static query failing.

An easy way to find where there's this problem in your code base is by adding this plugin — https://github.com/Urthen/case-sensitive-paths-webpack-plugin — as it enforces strict case sensitivity.

You can add it by:

`npm install case-sensitive-paths-webpack-plugin`

Then in your `gatsby-node.js` file, modify the webpack config to add the plugin:

```javascript
var CaseSensitivePathsPlugin = require('case-sensitive-paths-webpack-plugin');

exports.onCreateWebpackConfig = ({ actions, stage }) => {
  if (stage === "develop" || stage === "build-javascript") {
    actions.setWebpackConfig({
      plugins: [new CaseSensitivePathsPlugin()],
    });
  }
};
```

It'll then print errors to your terminal that look like the following:

<img width="893" alt="Screen Shot 2021-07-13 at 9 53 07 PM" src="https://user-images.githubusercontent.com/71047/125563370-972a347d-936b-4e4c-b12c-0e52ed32283f.png">

**Comment by dilatedigital** [comment]
Thank you, @KyleAMathews. This fixes my useStaticQuery error.

**Comment by ChildishhAlbino** [comment]
got told my issue (#31817) was a duplicate of this.

i've tried:
* using StaticQuery instead of `useStaticQuery` hook
* modifying my query to the simplest it can be
* ensuring all paths were lowercase
* completing reinstalling node dependencies.
* turning it on and off again
none of which have resolved my issue

anyone else in my situation?

**Comment by barbalex** [comment]
So I just updated my so far perfectly working and dev side tested site, this error appears and the site is DOWN 😭

**Comment by barbalex** [comment]
I added the CaseSensitivePathsPlugin as mentioned in https://github.com/gatsbyjs/gatsby/issues/26563#issuecomment-879585547 and it is outputting nothing.

**Comment by barbalex** [comment]
I remove my single `useStaticQuery` use (replaced the query result with static values) and now it works. Phew!!!!!!

**Comment by barbalex** [comment]
By the way: I am working on Windows, not OSX (though I have edited this project on OSX before).

**Comment by ScottGuymer** [comment]
I am seeing similar issues too. I am also on windows and seeing this on my local machine using `gatsby develop`. My colleague on OSX using the same code is not seeing the same issue.

I have tried all the workarounds and suggestions found here and everything seems to be up to date and correct.

Maybe a Windows specific thing?


I can see that in `gatsby-browser-entry.js` the `context` value is often an empty object. But sometimes if i trigger the query to change (by changing the name of the query) then the query id number will appear on `context` and the page will work.

Seems like it could be a caching issue somewhere?

**Comment by ScottGuymer** [comment]
Dropping into WSL and running this seems to confirm this is a Windows thing. When running from WSL it all works fine.

**Comment by jsg2021** [comment]
I'm on Fedora 34, I'm seeing this when running the build (both locally and on netlify), but not when running develop. I'm seeing an empty context object (assume it's the default global one defined when createContext is called)

Is there a context provider or some root node getting stomped on?

**Comment by Jassi10000** [comment]
Issue regarding the useStaticQuery #32825

Reproduction  :
After cloning and forking the repo
1. make setup-libs
2. make site

**Comment by AbdulSamadMalik** [comment]
### &nbsp; &nbsp;  Simple Solution
  1. Just capitalize the file name of the component in which you are using useStaticQuery hook.
    
     Example:   from **header.js** &nbsp;   **-->**   &nbsp;  ***Header.js***
 
  2. Then clear Gatsby cache and restart the server.
 
  3. This should fix the problem.

**Comment by ScottGuymer** [comment]
@AbdulSamadMalik this did not seems to solve the problem for me on windows.

**Comment by Jassi10000** [comment]
> @AbdulSamadMalik this did not seems to solve the problem for me on windows.
yes @AbdulSamadMalik  for me too it doesn't solve the problem

**Comment by AbdulSamadMalik** [comment]
> > @AbdulSamadMalik this did not seems to solve the problem for me on windows.
> > yes @AbdulSamadMalik  for me too it doesn't solve the problem

Well these are my dependencies make sure you have same versions as these

```json 
{
  "dependencies": {
    "gatsby": "^3.11.1",
    "gatsby-plugin-minify-classnames": "^0.3.1",
    "gatsby-plugin-react-helmet": "^4.12.0",
    "gatsby-plugin-sass": "^4.12.0",
    "gatsby-plugin-sitemap": "^4.8.0",
    "gatsby-source-filesystem": "^3.12.0",
    "gatsby-transformer-json": "^3.12.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-helmet": "^6.1.0",
    "sass": "^1.38.1"
 },
 "devDependencies": {
    "@types/react-helmet": "^6.1.2",
    "prettier": "2.3.2",
    "typescript": "^4.3.5"
  }
}
```

**Comment by wayward-man** [comment]
> I guess I figured it out. For some reason the problem was in `useStaticQuery` hook. I switched to `StaticQuery` component and problem is solved.

Could u show the usage of 'Statiquery' or the intro for it?

**Comment by kriscoleman** [comment]
We ran into this problem too, and nothing here seemed to fix it. 

We have decided as a team that we can't trust `useStaticQuery`. 

Our problem also came from the SEO component that we implemented which followed this doc: https://www.gatsbyjs.com/docs/add-seo-component/

We realized there really was no reason to *have* to use the useStaticQuery hook in the seo component, or any query for that matter. Instead we implemented it so that all the data needed for the helmet rendering is passed in via props. Now the SEO component doesn't care *how* you get your SEO data, and it's easy to test too. 

Now you can decide how to query for your SEO data however you want in your consumers. Since we're avoiding static queries, we just added the seo data to our pages query so it's pulled in along with the rest of the page data at build time. 

Works really well and is a lot simpler than using a query in the SEO component. 


Sadly, I don't have any answer for why the useStaticQuery hook seems to break infrequently and sometimes `gatsby clean` resolves it, but I do know that the query seems to query against our global context, and the behavior seems to suggest that sometimes it's undefined, and another developer above mentioned a few days ago they were able to capture the global context that's made in createContext sometimes being undefined, and they questioned if something (maybe a plugin?) is stomping on it sometimes, and I think that sounds really, really likely here.

**Comment by davwheat** [comment]
I'm now encountering this issue too.

- `useStaticQuery` is being called inside a component
- That component is used solely within a single MDX file
- Which is embedded into an MDX renderer inside a page template

This issue is happening on dev and prod, with Gatsby 4.

Repo with issue at specific commit hash: https://github.com/davwheat/personal-portfolio/tree/0ad42bf72eefb26dfd5432b99f903f8e7060a7c7

The troublesome code is here: https://github.com/davwheat/personal-portfolio/blob/0ad42bf72eefb26dfd5432b99f903f8e7060a7c7/blog/2021/10/04/CoverageResourceBlocksDiagram.tsx#L119-L126

...which is only referenced within MDX here: https://github.com/davwheat/personal-portfolio/blob/0ad42bf72eefb26dfd5432b99f903f8e7060a7c7/blog/2021/10/04/speed-differences-high-low-frequencies.mdx

All I wanted to do was reference an SVG file via its public URL instead of a data URI! 😅

**Comment by jsg2021** [comment]
https://github.com/jsg2021/gatsby-staticquery-error-repro this is my min reproduction... in gatsby 2 it's fine.

**Comment by samarthvats9** [comment]
Hey, I am new to this but I have tried this - `npm install react@^16.8.0 react-dom@^16.8.0` and it's working now. Please let me know if this is of any help.

**Comment by LekoArts** [comment]
This is a long opened issue that isn't actionable anymore as we merged PRs like https://github.com/gatsbyjs/gatsby/issues/33956 in the meantime and thus the overall environment has changed dramatically. If you still encounter this issue on Gatsby 4 and can provide a minimal reproduction, please open a new issue. (Comments on this one will be lost)

**Comment by jsg2021** [comment]
Its probably caused by the rss feed query requesting ['html' field ](https://www.gatsbyjs.com/plugins/gatsby-plugin-feed/#:~:text=frontmatter___date%5D%20%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20)%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20nodes%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20excerpt-,[html](https://www.gatsbyjs.com/plugins/gatsby-plugin-feed/#:~:text=frontmatter___date%255D%2520%257D%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520)%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520nodes%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520excerpt-,html,-fields%2520%257B%2520%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520slug%2520%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520frontmatter),-fields%20%7B%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20slug%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20frontmatter)

When I remove that part of the query, the error goes away.

**Comment by katiasmet** [comment]
It is still happening in Gatsby 4.

I don't use `gatsby-plugin-mdx`, `gatsby-plugin-feed` or `gatsby-plugin-offline`. It happens when I redeploy a build with changes to the query. It looks like something is cached in the browser because in incognito it works. Its is quite a big issue because when people revisit the website after the change they see a broken website..

I use useStaticQuery.

**Comment by jeustan** [comment]
Just trying to follow along with the tutorial and I'm stuck on this Unhandled Runtime Error. It does not occur on the Gatsby Cloud hosted version, however, it does occur locally. Tried incognito, doesn't work. Tried clean, clearing browser cache, reinstalling packages, scouring the code (most of it is copy pasted from the tutorial), nothing works.

I'm really stumped, by the tutorial. Wow.
